cmake/CPM.cmake

```diff
@@ -505,17 +505,45 @@ function(cpm_override_fetchcontent contentName)
   set_property(GLOBAL PROPERTY ${propertyName} TRUE)
 endfunction()
 
+macro(cpm_cmake_eval)
+  set(__ARGN "${ARGN}")
+  if(COMMAND cmake_language)
+    cmake_language(EVAL CODE "${__ARGN}")
+  else()
+    file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/eval.cmake "${__ARGN}")
+    include(${CMAKE_CURRENT_BINARY_DIR}/eval.cmake)
+  endif()
+endmacro()
+
 # Download and add a package from source
-function(CPMAddPackage)
+macro(CPMAddPackage)
+  set(__ARGN "${ARGN}")
+  list(LENGTH __ARGN __ARGN_Length)
+  if(__ARGN_Length EQUAL 1)
+    cpm_add_package_single_arg(${ARGN})
+  else()
+    # Forward preserving empty string arguments
+    # (https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4729)
+    set(__ARGN_Quoted)
+    foreach(__ARG IN LISTS __ARGN)
+      string(APPEND __ARGN_Quoted " [==[${__ARG}]==]")
+    endforeach()
+    cpm_cmake_eval("cpm_add_package_multi_arg( ${__ARGN_Quoted} )")
+  endif()
+endmacro()
+
+macro(cpm_add_package_single_arg arg)
   cpm_set_policies()
+  cpm_parse_add_package_single_arg("${arg}" __ARGN_multi)
 
-  list(LENGTH ARGN argnLength)
-  if(argnLength EQUAL 1)
-    cpm_parse_add_package_single_arg("${ARGN}" ARGN)
+  # The shorthand syntax implies EXCLUDE_FROM_ALL
+  list(APPEND __ARGN_multi EXCLUDE_FROM_ALL YES)
 
-    # The shorthand syntax implies EXCLUDE_FROM_ALL and SYSTEM
-    set(ARGN "${ARGN};EXCLUDE_FROM_ALL;YES;SYSTEM;YES;")
```

Comment:
Note: Change on this line was lost during merge - needs restore for `SYSTEM;YES`

Reply:
Done